features/000-anonymous-user/design.md

```diff
@@ -0,0 +1,112 @@
+# Anonymous User
+
+## Objective
+- Allow users to signup without using a real identity (e.g. login ID, OAuth).
+- Allow anonymous users to authenticate securely.
+- Allow promoting anonymous users to normal users.
+
+## Design
+
+### Anonymous identity
+
+A new identity type `anonymous` would be added. It is expected that users would
+either have no anonymous identity, or have exactly one anonymous identity
+without other types of identity.
+
+Anonymous identity has fields:
+- public key: the public key represented as a JWK.
+- key ID: a random string (`^[-\w]{8,64}$`), used for efficient lookup.
```

Comment:
The key ID is generated client side. Most likely it would be implemented as a UUID.

The key ID is in `kid` field of the JWK. You can see it in the example JWTs.